src/ray/raylet/node_manager.cc

```diff
@@ -2480,6 +2494,20 @@ void NodeManager::ForwardTaskOrResubmit(const Task &task,
 void NodeManager::ForwardTask(
     const Task &task, const ClientID &node_id,
     const std::function<void(const ray::Status &, const Task &)> &on_error) {
+  // Override spillback for direct tasks.
+  if (task.OnSpillback() != nullptr) {
+    GcsNodeInfo node_info;
+    bool found = gcs_client_->client_table().GetClient(node_id, &node_info);
+    if (!found) {
+      RAY_LOG(INFO) << "No node manager client found for GCS client id " << node_id;
+      task.OnSpillback()(ClientID::Nil(), "", 0);
```

Comment:
When would this happen? What does the client do in this case?

Reply:
Hmm I think this would have to be a bug. I may just change this to RAY_CHECK.